InnerEye/Azure/azure_runner.py

```diff
@@ -273,40 +274,47 @@ def get_or_create_python_environment(azure_config: AzureConfig,
     return env
 
 
-def get_dataset_consumption(azure_config: AzureConfig, azure_dataset_id: str) -> DatasetConsumptionConfig:
+def get_dataset_consumption(azure_config: AzureConfig,
+                            azure_dataset_id: str,
+                            dataset_index: int = 0) -> DatasetConsumptionConfig:
     """
     Creates a configuration for using an AzureML dataset inside of an AzureML run. This will make the AzureML
     dataset with given name available as a named input, using INPUT_DATA_KEY as the key.
     :param azure_config: azure related configurations to use for model scale-out behaviour
     :param azure_dataset_id: The name of the dataset in blob storage to be used for this run. This can be an empty
     string to not use any datasets.
+    :param dataset_index: suffix for the dataset name, dataset name will be set to INPUT_DATA_KEY_idx
     """
     azureml_dataset = get_or_create_dataset(azure_config, azure_dataset_id=azure_dataset_id)
     if not azureml_dataset:
         raise ValueError(f"AzureML dataset {azure_dataset_id} could not be found or created.")
-    named_input = azureml_dataset.as_named_input(INPUT_DATA_KEY)
+    named_input = azureml_dataset.as_named_input(f"{INPUT_DATA_KEY}_{dataset_index}")
     return named_input.as_mount() if azure_config.use_dataset_mount else named_input.as_download()
 
 
 def create_run_config(azure_config: AzureConfig,
                       source_config: SourceConfig,
                       azure_dataset_id: str = "",
+                      extra_azure_dataset_ids: List[str] = [],
                       environment_name: str = "") -> ScriptRunConfig:
     """
     Creates a configuration to run the InnerEye training script in AzureML.
     :param azure_config: azure related configurations to use for model scale-out behaviour
     :param source_config: configurations for model execution, such as name and execution mode
     :param azure_dataset_id: The name of the dataset in blob storage to be used for this run. This can be an empty
     string to not use any datasets.
+    :param extra_azure_dataset_ids: List of extra datasets in blob storage to be used for this run. This can be empty.
     :param environment_name: If specified, try to retrieve the existing Python environment with this name. If that
     is not found, create one from the Conda files provided in `source_config`. This parameter is meant to be used
     when running inference for an existing model.
     :return: The configured script run.
     """
-    if azure_dataset_id:
-        dataset_consumption = get_dataset_consumption(azure_config, azure_dataset_id)
-    else:
-        dataset_consumption = None
+    dataset_consumptions = {}
+    all_dataset_ids = [azure_dataset_id] + extra_azure_dataset_ids
```

Comment:
If we are merging the dataset ids, why don't we use a single variable for all? It's not clear to me what the need for two separate variables is?

Reply:
Because the InnerEye container expects this to be a string, not a list. It would break backwards compatibility in multiple places if we merge both arguments. And we don't really want to a list of dataset for InnerEyeContainers

Reply:
Makes sense =)